### PR TITLE
Increase coverage

### LINKED ISSUE #102: Increase code coverage to 70%
Minimal code coverage needed for the project.

### PR DESCRIPTION
**Pull request by theodoreleebrant**
# Increase test coverage for code

## Summary
I have increased the test coverage by the following:
* Changing the codecov YAML file to exclude nontestable UI classes
* Implementing new tests for StudentRecord (0% -> 100%)
* ~~Implementing new tests for StudentRecordCard (0% -> 100%)~~ Cancelled, JavaFX borked up
* Implementing one test for non-default constructor in ModelManager

Motivation: Code coverage is an important metric, and those that cannot be tested should not be put into the metric as it would cloud the true percentage. For those which should have tests, I am trying to implement more tests.

No dependency change.
Progress for issue #102 but not closing it

## Type of change
<Select all that apply, in the form [x]>
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing
Testing was done via JUnit. StudentRecordCardTest uses a JavaFX initialisation.

**Test Configuration**: Junit 5
* OS: Ubuntu 20.04 LTS

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/seedu/taskmaster/model/student/Name.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
      * @param name A valid name.
      */
     public Name(String name) {
+        assert true;
```
Comment (jaya-r3) [inline src/main/java/seedu/taskmaster/model/student/Name.java]: Please consider removing this line

**Review by jaya-r3 — COMMENTED** [review]
Looks good, can be merged once `assert true` is removed from Name.java

File: src/main/java/seedu/taskmaster/model/student/Name.java (hunk 1 above)
Comment (theodoreleebrant) [reply src/main/java/seedu/taskmaster/model/student/Name.java]: Done

**Review by jaya-r3 — APPROVED** [review]
LGTM

**Pull request merged**